FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -128,50 +136,70 @@ - (void)signInTesterWithCompletion:(void (^)(NSError *_Nullable error))completio
 }
 
 - (void)signOutTester {
+  NSError *error;
+  BOOL didClearAuthState = [FIRAppDistributionAuthPersistence clearAuthState:&error];
+  // TODO (schnecle): Add in FIRLogger to report when we have failed to clear auth state
+  if (!didClearAuthState) {
+    NSLog(@"Error clearing token from keychain: %@", [error localizedDescription]);
+  }
+
   self.authState = nil;
   self.isTesterSignedIn = false;
 }
 
 - (void)fetchReleases:(FIRAppDistributionUpdateCheckCompletion)completion {
-  NSURLSession *URLSession = [NSURLSession sharedSession];
-  NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
-  NSString *URLString =
-      [NSString stringWithFormat:kReleasesEndpointURL, [[FIRApp defaultApp] options].googleAppID];
-  [request setURL:[NSURL URLWithString:URLString]];
-  [request setHTTPMethod:@"GET"];
-  [request setValue:[NSString
-                        stringWithFormat:@"Bearer %@", self.authState.lastTokenResponse.accessToken]
-      forHTTPHeaderField:@"Authorization"];
-
-  NSURLSessionDataTask *listReleasesDataTask = [URLSession
-      dataTaskWithRequest:request
-        completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-          if (error) {
-            // TODO: Reformat error into error code
-            completion(nil, error);
-            return;
-          }
-
-          NSHTTPURLResponse *HTTPResponse = (NSHTTPURLResponse *)response;
-
-          if (HTTPResponse.statusCode == 200) {
-            [self handleReleasesAPIResponseWithData:data completion:completion];
-          } else {
-            // TODO: Handle non-200 http response
-            @throw([NSException exceptionWithName:@"NotImplementedException"
-                                           reason:@"This code path is not implemented yet"
-                                         userInfo:nil]);
-          }
-        }];
-
-  [listReleasesDataTask resume];
+  [self.authState performActionWithFreshTokens:^(NSString *_Nonnull accessToken,
+                                                 NSString *_Nonnull idToken,
+                                                 NSError *_Nullable error) {
+    if (error) {
+      // TODO (schnecle): Add in FIRLogger log statement
+      NSLog(@"Error fetching fresh tokens: %@", [error localizedDescription]);
+      [self signOutTester];
+      return;
+    }
+
+    // perform your API request using the tokens
+    NSURLSession *URLSession = [NSURLSession sharedSession];
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+    NSString *URLString =
+        [NSString stringWithFormat:kReleasesEndpointURL, [[FIRApp defaultApp] options].googleAppID];
```

Comment:
Recommend grabbing and holding on to the `googleAppID` in your SDK's initializer `initWithApp`. Seems to be best practice and I believe defaultApps can be deleted.

Reply:
We're going to address this in a future PR, as it is not actually related to the change for auth persistence.